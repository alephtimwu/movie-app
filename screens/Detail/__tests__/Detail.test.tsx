import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
// import { CrpAll } from '__mocks__';
import mockMovie from '../../../constants/movies';
import GlobalState from '../../../store';
import DetailScreen from '../index';

const Stack = createStackNavigator();

const theme = {
  dark: true,
  colors: {
    primary: 'rgb(255, 45, 85)',
    background: '#1d1d27',
    card: 'rgb(255, 255, 255)',
    text: '#ffffff',
    border: 'rgb(199, 199, 204)'
  }
};

describe('DetailScreen =>', () => {
  let tree = null;
  const handlePress = jest.fn();

  beforeEach(() => {
    tree = render(
      <GlobalState.Provider>
        <NavigationContainer theme={theme}>
          <Stack.Navigator initialRouteName="Detail">
            <Stack.Screen
              name="Detail"
              component={DetailScreen}
              options={({ navigation, route }) => ({
                title: '',
                headerTransparent: true,
                headerStyle: {
                  backgroundColor: '#1D1D27',
                  borderWidth: 0,
                  borderColor: '#1D1D27'
                },
                headerTintColor: '#fff'
              })}
            />
          </Stack.Navigator>
        </NavigationContainer>
      </GlobalState.Provider>
    );
  });

  test('DetailScreen should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  //   test('DetailScreen should render correctly with custom width height', () => {
  //     const { rerender, getByTestId } = tree;
  //     rerender(
  //       <DetailScreen
  //         movie={mockMovie.mock[0]}
  //         onPress={handlePress}
  //         width={80}
  //         height={100}
  //       />
  //     );
  //     fireEvent.press(getByTestId('card-press'));
  //     expect(handlePress).toBeCalled();
  //   });
});
