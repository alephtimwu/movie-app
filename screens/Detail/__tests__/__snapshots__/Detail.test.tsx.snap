// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailScreen => DetailScreen should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "#1d1d27",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Object {
                          "backgroundColor": "#1d1d27",
                          "flex": 0,
                        }
                      }
                    />
                    <RCTSafeAreaView
                      emulateUnlessSupported={true}
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "#1d1d27",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "flexShrink": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "flex": 1,
                                "justifyContent": "center",
                              },
                              Object {
                                "flexDirection": "row",
                                "justifyContent": "space-around",
                                "padding": 10,
                              },
                            ]
                          }
                        >
                          <ActivityIndicator
                            animating={true}
                            color="#fff"
                            hidesWhenStopped={true}
                            size="large"
                          />
                        </View>
                      </View>
                    </RCTSafeAreaView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 18,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
